sdk/cpp/core/src/common_utilities.cpp

```diff
@@ -0,0 +1,143 @@
+// YANG Development Kit
+// Copyright 2016 Cisco Systems. All rights reserved
+//
+////////////////////////////////////////////////////////////////
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+//////////////////////////////////////////////////////////////////
+
+#include "common_utilities.hpp"
+#include "entity_data_node_walker.hpp"
+#include "xml_subtree_codec.hpp"
+
+using namespace std;
+using namespace ydk;
+
+string trim(const string& str)
+{
+    const string whitespace = " \t\n\r";
+    const auto strBegin = str.find_first_not_of(whitespace);
+    if (strBegin == string::npos)
+        return ""; // no content
+
+    const auto strEnd = str.find_last_not_of(whitespace);
+    const auto strRange = strEnd - strBegin + 1;
+
+    return str.substr(strBegin, strRange);
+}
+
+bool replace(string& subject, const string& search, const string& replace)
+{
+    size_t pos = 0;
+    int replace_count = 0;
+    while ((pos = subject.find(search, pos)) != string::npos)
+    {
+         subject.replace(pos, search.length(), replace);
+         pos += replace.length();
+         replace_count++;
+    }
+    return replace_count>0;
+}
+
+bool has_xml_escape_sequences(const string& xml)
+{
+    if (xml.find("&lt;") != string::npos  ||
+        xml.find("&gt;") != string::npos  ||
+        xml.find("&amp;") != string::npos ||
+        xml.find("&quot;") != string::npos)
+    {
+        return true;
+    }
+    return false;
+}
+
+string replace_xml_escape_sequences(const string& xml)
```

Comment:
According to [the codecov report](https://codecov.io/gh/CiscoDevNet/ydk-gen/pull/713/diff#D23-69), this function is not being tested. Can you please add a test case for this?